liquidity-rewards/test/UniswapPoolRewardsTest.js

```diff
@@ -1,27 +1,276 @@
     });
 
-    it('deployed successfully', async function() {
-        const address = await this.contract.address;
-        assert.notEqual(address, 0x0);
-        assert.notEqual(address, "");
-        assert.notEqual(address, null);
-        assert.notEqual(address, undefined);
+    it('Two depositors with different (1:3) deposits wait for DURATION and DURATION/2', async function () {
+      //
+      // 1x: +--------------+
+      // 3x:      +---------+
+      //
+
+      const deposit1 = new BN(web3.utils.toWei('1'));
+      await this.pool.deposit(deposit1, { from: wallet1 });
+      const depositTime1 = await time.latest();
+
+      await time.increaseTo(depositTime1.add(AMONTH.div(new BN(3))));
+
+      const deposit2 = new BN(web3.utils.toWei('3'));
+      await this.pool.deposit(deposit2, { from: wallet2 });
+      const depositTime2 = await time.latest();
+
+      const timeDiff = depositTime2.sub(depositTime1);
+      const rewardPerToken1 = this.rewardRate.mul(timeDiff).mul(_1e18).div(deposit1);
+      expect(await this.pool.rewardPerToken()).to.be.bignumber.almostEqualDiv1e18(rewardPerToken1);
+      expect(await this.pool.rewardsEarned(wallet1)).to.be.bignumber.almostEqualDiv1e18(rewardPerToken1.mul(deposit1).div(_1e18));
+      expect(await this.pool.rewardsEarned(wallet2)).to.be.bignumber.equal('0');
+
+      // Forward to week 3 and notifyReward weekly
+      await time.increase(AMONTH.mul(new BN(2)).div(new BN(3)));
+
+      const rewardPerToken2 = this.rewardRate.mul(AMONTH.sub(timeDiff)).mul(_1e18).div(deposit1.add(deposit2));
+      const rewardPerToken = rewardPerToken1.add(rewardPerToken2);
+      expect(await this.pool.rewardPerToken()).to.be.bignumber.almostEqualDiv1e18(rewardPerToken);
+      expect(await this.pool.rewardsEarned(wallet1)).to.be.bignumber.almostEqualDiv1e18(rewardPerToken1.add(rewardPerToken2).mul(deposit1).div(_1e18));
+      expect(await this.pool.rewardsEarned(wallet2)).to.be.bignumber.almostEqualDiv1e18(rewardPerToken2.mul(deposit2).div(_1e18));
+      console.log("------------------------------------------------------------------------");
+      console.log("(calculated for a duration of one month)");
+      console.log("rewardRate:", this.rewardRate.toString());
+      console.log("rewardPerToken calc:", rewardPerToken.div(_1e18).toString());
+      console.log("rewardPerToken actual:",(await this.pool.rewardPerToken()).div(_1e18).toString());
+      console.log("------------------------------------------------------------------------"); 
+      console.log("rewardsEarned wallet1:", (await this.pool.rewardsEarned(wallet1)).div(_1e18).toString());
+      console.log("rewardsEarned wallet2:", (await this.pool.rewardsEarned(wallet2)).div(_1e18).toString());
+      console.log("------------------------------------------------------------------------");
     });
 
-    // TODO
+    it('Three depositors with different (1:3:5) deposits waits for different DURATION', async function () {
```

Comment:
Does `1:3:5` mean that one account deposits 1 `$ELIMU`, another deposits 5 `$ELIMU`, and another deposits 5 `$ELIMU`?

Reply:
Yes, that's the idea!